ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/TransactionCommitter.scala

```diff
@@ -614,4 +631,32 @@ private[kvutils] object TransactionCommitter {
   def getContractState(commitContext: CommitContext, key: DamlStateKey): DamlContractState =
     commitContext.get(key).getOrElse(throw Err.MissingInputState(key)).getContractState
 
+  @SuppressWarnings(Array("org.wartremover.warts.Option2Iterable"))
+  private def transactionMinRecordTime(
+      submissionTime: Instant,
+      ledgerTime: Instant,
+      maybeDeduplicateUntil: Option[Instant],
+      maxSkew: Duration): Instant =
+    List(
+      maybeDeduplicateUntil
+        .map(_.plus(TimeModel.resolution)), // DeduplicateUntil defines a rejection window, endpoints inclusive
+      Some(TimeModel.minRecordTime(ledgerTime, maxSkew)),
+      Some(submissionTime.minus(maxSkew))
```

Comment:
Minor: I'd use the `TimeModel` methods here as well for uniformity, although it's not required.